### PR TITLE
Feature sb 22 add sign up fe

### PR DESCRIPTION
**Pull request by ValentinaTenus**
1. created reusable components (BaseButton with custom variants and CustomFormGroup)
2. auth slice and auth api
3. baseQuery and baseQueryWithReauth for managing queries with token and without
4. auth page 
5. sign-up form  with useSignUpForm hook for logic
6. validation according to requeres
7. added fonts, color, button and typography variants to theme
8. added some changes in eslint, typescript configurations(for instanse, run eslint before commit)
sign-up-form-small

sign-up-page

*  bottom links aligned

### CODE REVIEW

File: src/components/auth/google-button/google-button.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import styles from "./styles.module.css";
+
+const GoogleButton: React.FC = () => {
+	return (
+		<Button
+			fullWidth
+			sx={{
+				"&:hover": { backgroundColor: "#EDEAE9" },
+				backgroundColor: "#EDEAE9",
```
Comment (lyudad) [inline src/components/auth/google-button/google-button.tsx]: 1.3.2

File: src/components/auth/sign-in-form/sign-in-form.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const SignInForm: React.FC = () => {
+	return <>Log in</>;
```
Comment (lyudad) [inline src/components/auth/sign-in-form/sign-in-form.tsx]: 1.3.3

File: src/components/auth/sign-up-form/sign-up-form.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+						error={errors.email}
+						label="Email"
+						name="email"
+						placeholder="your_email@gmail.com"
+						type="email"
+					/>
+					<CustomFormGroup
+						control={control}
+						error={errors.password}
+						label="Password*"
```
Comment (lyudad) [inline src/components/auth/sign-up-form/sign-up-form.tsx]: 1.3.3

File: src/components/common/base-button/base-button.tsx
Comment (lyudad) [inline src/components/common/base-button/base-button.tsx:1]: rename to index.tsx

File: src/libs/helpers/storage.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+const storageKey = {
```
Comment (lyudad) [inline src/libs/helpers/storage.ts]: I assume redux-persist library will replace this

File: src/components/auth/google-button/google-button.tsx (hunk 1 above)
Comment (ValentinaTenus) [reply src/components/auth/google-button/google-button.tsx]: moved to constants

File: src/components/auth/sign-in-form/sign-in-form.tsx (hunk 2 above)
Comment (ValentinaTenus) [reply src/components/auth/sign-in-form/sign-in-form.tsx]: done

File: src/components/auth/sign-up-form/sign-up-form.tsx (hunk 3 above)
Comment (ValentinaTenus) [reply src/components/auth/sign-up-form/sign-up-form.tsx]: done

File: src/components/common/base-button/base-button.tsx
Comment (ValentinaTenus) [reply src/components/common/base-button/base-button.tsx:1]: done

File: src/libs/helpers/storage.ts (hunk 4 above)
Comment (ValentinaTenus) [reply src/libs/helpers/storage.ts]: replaced storage with redux-persist

**Comment by ValentinaTenus** [comment]
1. I've fixed your remarks
2. added private and public routes
3. saved user data and tokens after sign up(we will change it after adding email verification)

File: src/components/auth/auth-links/auth-links.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+				color="primary.main"
+				component="span"
+				ml="10px"
+				variant="playfairDisplay"
+			>
+				{t("SignUpComponent.signIn")}
+			</Typography>
+		</Link>
+	);
+};
```
Comment (lyudad) [inline src/components/auth/auth-links/auth-links.tsx]: there are 2 components in one file, never do it

Comment (ValentinaTenus) [reply src/components/auth/auth-links/auth-links.tsx]: done

**Pull request merged**